### PR TITLE
migrate simple prometheus receiver to prometheus receiver

### PR DESCRIPTION
**Pull request by chadpatel**
**Description:** 
Minor refactor to use the more configurable `prometheusreceiver` instead of `simpleprometheusreceiver`, this allows us to use prometheus relabel rules to filter what we want instead of pushing this extra data through the OTel receiver/processor pipeline

Using the processor-less method, we cannot filter out the prometheus metadata metrics.  Once per minute a log message like this will be published:
Screen Shot 2023-07-17 at 4 43 33 PM

This is a two way door, we can always move the relabeling into a processor

My initial goal was to improve network RX traffic but ultimately that is not possible because the /metrics endpoint does not offer server-side metric filtering.  (i.e. `Params` - https://prometheus.io/docs/prometheus/latest/federation/)

**Testing:** 
I confirmed the data is still flowing in my test cluster
Screen Shot 2023-07-17 at 4 42 35 PM
Screen Shot 2023-07-17 at 4 42 17 PM

Small sample set (~3 hours)
The new approach uses less memory and CPU
Screen Shot 2023-07-18 at 12 39 34 PM

**Documentation:** N/A

### CODE REVIEW

**Comment by chadpatel** [comment]
I pushed a new commit which should fix the "checks" failure

File: receiver/awscontainerinsightreceiver/internal/k8sapiserver/prometheus_scraper.go [hunk 1]
```diff
@@ -19,74 +19,150 @@ import (
...
+				CAFile:             caFile,
 				InsecureSkipVerify: false,
 			},
 		},
-		MetricsPath:        "/metrics",
-		CollectionInterval: collectionInterval,
-		UseServiceAccount:  true,
+		ScrapeInterval:  model.Duration(collectionInterval),
+		ScrapeTimeout:   model.Duration(collectionInterval),
+		JobName:         fmt.Sprintf("%s/%s", "containerInsightsKubeAPIServerScraper", opts.Endpoint),
```
Comment (SaxyPandaBear) [inline receiver/awscontainerinsightreceiver/internal/k8sapiserver/prometheus_scraper.go:107]: super minor nit - the prefix for this job name could be pulled out as a const

**Review by SaxyPandaBear — APPROVED** [review]

File: receiver/awscontainerinsightreceiver/internal/k8sapiserver/prometheus_scraper.go [hunk 2]
```diff
@@ -19,74 +19,150 @@ import (
...
+		HonorTimestamps: true,
+		Scheme:          "https",
+		MetricsPath:     "/metrics",
+		ServiceDiscoveryConfigs: discovery.Configs{
+			&discovery.StaticConfig{
+				{
+					Targets: []model.LabelSet{
```
Comment (sky333999) [inline receiver/awscontainerinsightreceiver/internal/k8sapiserver/prometheus_scraper.go:114]: Dont need to change now, but if we see this config growing big in the future, we could actually keep this in a yaml and embed it, deserialize that into a config and set some of the properties that exist here in code as overrides.

Comment (sky333999) [reply receiver/awscontainerinsightreceiver/internal/k8sapiserver/prometheus_scraper.go:114]: Could also do placeholders in the yaml and once we read the embed string, replace placeholders with actual values. For ex, have a placeholder like `<ClusterName>` which gets replaced with the actual cluster name in code.

**Review by sky333999 — APPROVED** [review]

**Comment by sky333999** [comment]
Appreciate all the test results in the Overview and its a great win to gain on the cpu & mem while cleaning up the code!

File: receiver/awscontainerinsightreceiver/internal/k8sapiserver/prometheus_scraper.go (hunk 2 above)
Comment (chadpatel) [reply receiver/awscontainerinsightreceiver/internal/k8sapiserver/prometheus_scraper.go:114]: That makes sense.  I think once we get to two scrape sources then we should refactor

**Pull request merged**